### ISSUE TITLE
WCAG improvement of SearchSelect component (violation)

### ISSUE THREAD

**Issue by Maren-SSB** [PxTools/pxtools-project#13]
SearchSelect-komponenten har to brudd:

### 1.  Den åpner seg automatisk ved fokus

SearchSelect-komponenten (År-filter på startsiden) åpner seg automatisk ved fokus. Dette kan være et brudd på WCAG 3.2.1, siden komponenten endrer tilstand uten at brukeren har initiert en handling. Komponenter bør kun åpne eller endre tilstand ved eksplisitt brukerhandling (f.eks. enter, space eller piltaster), ikke ved fokus alene.

Endringen i SearchSelect blir da at bruker må enten trykke på enter, space eller pil ned for å åpne nedtrekkslisten.

<br>

### 2. Utilstrekkelig fokusmarkering

Valgene i nedtrekkslisten mangler fokusmarkering (den bruker kun hover):

<img width="463" height="461" alt="Image" src="https://github.com/user-attachments/assets/10813dcf-57f0-48cd-9bce-3800baa5d8e2" />

<br><br>

Ved tastaturnavigering skal den lilla fokusmarkeringen brukes (3 px). Da vil selve inputfeltet ikke lenger ha fokus, siden fokuset er i nedtrekkslisten:

<br>

<img width="1086" height="882" alt="Image" src="https://github.com/user-attachments/assets/96284389-ef2c-4c8f-a98b-c7d07ceccec0" />

<br><br>

Det er også viktig at det scrolles nedover i listen når en navigerer nedover med tastatur, slik at fokusmarkeringen ikke forsvinner.

**Comment by SjurSutterudSagen** [comment]
I also ran into this issue: https://github.com/PxTools/PxWeb2/issues/1290

**Comment by SjurSutterudSagen** [comment]
Might have found another issue here @Maren-SSB: I can't seems to use the clear button/icon for the search, when using a keyboard. Is that intentional?

Also, how is 1 going to affect searching in the filter?

**Comment by Maren-SSB** [comment]
@SjurSutterudSagen @PerIngeVaaje Godt funn, Sjur! Det med at en ikke kan bruke clear button.

Basically er dette den samme komponenten som Designsystemet har: https://designsystemet.no/no/components/docs/suggestion/overview. Den skal fungere likt.

Punkt 1 i oppgaven påvirker ikke søket på noen måte. Den skal åpne seg når en begynner å skrive noe. Eller om en trykker på den med musepeker. Men om en navigerer seg til komponenten med tastatur, skal den ikke åpnes automatisk

**Comment by PerIngeVaaje** [comment]
@Maren-SSB Has refactored the component. Now it doesn't open automatically when tabbed to. And the clear button works even on non matched search.
Test branch: https://fix-pxweb-13-wcag-improvemen.pxweb2.pages.dev/en/

**Comment by Maren-SSB** [comment]
@PerIngeVaaje Meget nice! Oppdaget to mindre problemer:

- Jeg får ikke bruke enter på x-ikonet, kun space. Enter må også kunne brukes
- Space skal kunne brukes for å åpne nedtrekkslisten, men nå blir dette oppfattet som en input og gir meg derfor meldingen «No results». Du kan se hvordan Designsystemet har løst det. Men akkurat dette siste punktet er ikke så viktig, så ikke bruk tid på det om det er problematisk

**Comment by PerIngeVaaje** [comment]
@Maren-SSB Hmm. Hos meg fungerer enter på x-ikonet. ny Mac?

**Comment by Maren-SSB** [comment]
@PerIngeVaaje Ja, det må være den nye macen, eller et behov for mer ferie, for nå virket det fint hos meg også. Så det er kun punkt to i den kommentaren min som er aktuelt da 🥳

**Comment by KentMossback** [comment]
@PerIngeVaaje Test

**Comment by Maren-SSB** [comment]
@PerIngeVaaje Nå fungerer det strålende! ✅ ✅ ✅
